### PR TITLE
Add util functions for backendConfig annotation

### PR DESCRIPTION
**Pull request by MrHohn**
/assign @nicksardo @bowei

### CODE REVIEW

**Comment by nicksardo** [comment]
LGTM, will let bowei@ verify annotation value structure is as desired.

**Comment by MrHohn** [comment]
Revised and added `ports` as a top level field per discussion with @bowei. PTAL, thanks!

File: pkg/annotations/service.go [hunk 1]
```diff
@@ -40,6 +40,15 @@ const (
 	// 3. Adding this annotation on ingress.
 	NetworkEndpointGroupAlphaAnnotation = "alpha.cloud.google.com/load-balancer-neg"
 
+	// BackendConfigKey is a stringified JSON with two fields:
+	// - "ports": a map of port names to backendConfig names
```
Comment (nicksardo) [inline pkg/annotations/service.go]: port names or port numbers? Or just port names?

Comment (MrHohn) [reply pkg/annotations/service.go]: Thanks, changed to "port names or port numbers".

**Pull request merged**